docs/src/extend/custom-formatters.md

```diff
@@ -101,21 +87,23 @@ The `results` object passed into a formatter is an array of objects containing t
 ]
 ```
 
-### The `result` Object
+The remainder of this section contains reference information on how to work with custom formatter functions.
+
+#### The `result` Object
 
 <!-- This section is copied from the "Node.js API" page. Changes to this section should
 also be manually applied to that page. -->
 
 Each object in the `results` array is a `result` object. Each `result` object contains the path of the file that was linted and information about linting issues that were encountered. Here are the properties available on each `result` object:
 
 * **filePath**: The absolute path to the file that was linted.
-* **messages**: An array of `message` objects. See below for more info about messages.
+* **messages**: An array of [`message`](#the-message-object) objects. See below for more info about messages.
 * **errorCount**: The number of errors for the given file.
 * **warningCount**: The number of warnings for the given file.
 * **source**: The source code for the given file. This property is omitted if this file has no errors/warnings or if the `output` property is present.
 * **output**: The source code for the given file with as many fixes applied as possible. This property is omitted if no fix is available.
 
-### The `message` Object
+##### The `message` Object
```

Comment:
We are getting pretty deep into heading levels here. Is this correct?

Reply:
i don't think it's *necessary* but it makes hierarchical sense since it's just used in the `results` object, which is an H4. 

happy to keep this as H4 if you'd like tho.

Reply:
Okay, it looks a bit weird to me, but not a big enough deal to hold this up.